keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
-1. `policyActionsConfigMapData`: encoded data, known to the driver, and defined by the admin when creating a `BucketAccessClass`. Contains a set of provisioner/platform defined policy actions to a given user identity. Contents of the ConfigMap that the *policyActionsConfigMap* field in the `BucketAccessClass` refers to. A sample value of this field could look like:
-```
-   {“Effect”:“Allow”,“Action”:“s3:PutObject”,“Resource”:“arn:aws:s3:::profilepics/*“},
-   {“Effect”:“Allow”,“Action”:“s3:GetObject”,“Resource”:“arn:aws:s3:::profilepics/*“},
-   {“Effect”:“Deny”,“Action”:“s3:*“,”NotResource”:“arn:aws:s3:::profilepics/*“}
-```
-1. `principal`: username/access-key for the object storage provider to uniquely identify the user who has access to this bucket. This value is minted by the provisioner (and set in the BA) when the `BucketAccessRequest` omits a ServiceAccount. There are several use cases involving `principal`: 
-   - Not relevant if serviceAccount is supplied.
-   - For brownfield cases when there is already a storage-side principal that you want to use for access, the admin can supply this, the driver will (possibly) enable access for that principal to the bucket in question and will return credentials.
-   - For greenfield cases, this could be populated by the driver. Maybe so that the driver can request deletion of this principal during deprovisioning
-1. `parameters`:  A map of string:string key values.  Allows admins to control user and access provisioning by setting provisioner key-values. Copied from `BucketAccessClass`. 
-1. `accessGranted`: if true access has been granted to the bucket. If false then access to the bucket has not been granted.
-
-#### BucketAccessClass
-
-An immutable, cluster-scoped, custom resource providing a way for admins to specify policies that may be used to access buckets. A `BucketAccessClass` (BAC) is required for both greenfield and brownfield use cases. Besides naming the provisioner, a BAC references a config map which is expected to define the access policy for a given provider. It is typical for these config maps to reside in each provisioner's namespace, though this is not required.  Unlike BucketClasses, there is no default BAC.
+Here is a TL;DR version:
 
-```yaml
-apiVersion: cosi.io/v1alpha1
-kind: BucketAccessClass
-metadata: 
-  name:
-provisioner: [1]
-policyActionsConfigMap: [2]
-  name:
-  namespace:
-parameters: [3]
+ - BucketClaims/Bucket are similar to PVC/PV. 
+ - BucketClaim is used to request generation of new buckets/binding to existing ones. 
+ - Buckets represent the actual Bucket. 
+ - BucketClass is similar to StorageClass. It is meant for admins to define and control policies for Bucket Creation
+ - BucketAccess is required before a bucket can be "attached" to a pod.
```

Comment:
will the wording `a bucket be "attached" to a pod.` creates confusion for the user, actually secret created as part of `bucketaccess` attach to pod right?